gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageOptions.java

```diff
@@ -247,6 +262,15 @@ public GoogleCloudStorageOptions build() {
       checkArgument(
           (instance.getProxyUsername() == null) == (instance.getProxyPassword() == null),
           "both proxyUsername and proxyPassword should be null or not null together");
+      checkArgument(
+          (instance.getEncryptionAlgorithm() != null
```

Comment:
You can simplify this with XOR operator:
```java
checkArgument(
    !(instance.getEncryptionAlgorithm() == null
        ^ instance.getEncryptionKey() == null
        ^ instance.getEncryptionKeyHash() == null),
    "encryptionAlgorithm, encryptionKey and encryptionKeyHash should be null or not null"
              + " together");
```

Reply:
This actually won't work because true ^ true = false and then false ^ true = true and then !true = false which is wrong.

Remain the same logic, might not seem most concise but clear enough.